synapse/replication/http/_base.py

```diff
@@ -433,7 +433,7 @@ async def _check_auth_and_handle(
 
         if self.WAIT_FOR_STREAMS:
             response[_STREAM_POSITION_KEY] = {
-                stream.NAME: stream.current_token(self._instance_name)
+                stream.NAME: stream.minimal_local_current_token()
```

Comment:
Can you remind me: why does the requester submit this information again? When submitted, should the requester submit the minimum or maximum current token?

Reply:
This data is used so that the other side can "wait" until all the streams have advanced to the position included, i.e. when the instance starts processing the request/response it knows that e.g. all cache invalidations have correctly happened.

The difference between request/response here is fairly immaterial.